content/chainguard/fips/not-approved-algorithms.md

```diff
@@ -0,0 +1,170 @@
+
+Alternatively submit upstream fixes to correct such usage, here are some examples contributed by Chainguardians:
+* Google Cloud Platform storage [pull request](https://github.com/googleapis/python-storage/pull/1522)
+* PyPDF project [pull requst](https://github.com/py-pdf/pypdf/pull/3438)
+
+If there are python projects that need MD5 access in FIPS mode and currently do not use `usedforsecurity=False` please open a support request for Chainguard engineering to look into implementing.
+
+### Python pyca/cryptography FIPS and MD5
+
+If one self-builds pycryptography or installs it from PyPI, it may come with vendored and statically linked copy of cryptographic library and may not operate in FIPS mode. In such cases, all usage of it may be not-approved.
+
+If one installs pycryptography through custom assembly, it will be linked with Chainguard OpenSSL and fill correctly enforce FIPS compliance and will operate in approved mode.
+
+### .NET FIPS and MD5
+
+.NET upstream chooses to always fetch MD5 algorithm with `-fips` property query string, thus to verify that MD5 is blocked for security purposes, one has to test calcualting HMAC-MD5 or attempting to create or verify a digital signature with MD5.
+
+See dotnet/runtime [source code](https://github.com/dotnet/runtime/blob/25800e6537cd47a0a0533fb63bb0fa60d600ec45/src/native/libs/System.Security.Cryptography.Native/pal_evp.c#L290)
+
+### Go FIPS and MD5
+
+MD5 was deprecated before golang language was created. Hence the MD5 implementation at [crypto/md5](https://pkg.go.dev/crypto/md5) states that it should not be used for security purposes.
+
+Currently Chainguard offers two Golang FIPS toolchains. Both of them are based on [microsoft/go](https://github.com/microsoft/go) toolchain and use OpenSSL FIPS at runtime:
+
+* The [go-fips](https://images.chainguard.dev/directory/image/go-fips/overview) always allows MD5 usage.
+
+* The [go-msft-fips](https://images.chainguard.dev/directory/image/go-msft-fips/overview) always blocks MD5 usage.
+
+There are a few codepaths in golang standard library where MD5 is used for authentication. Chainguard is working to correctly block such usage, and unify the above toolchains into one.
+
+### Java FIPS and MD5
+
+Currently all Java FIPS implementations in Chainguard FIPS images is powered by BouncyCastle FIPS as jar on module path. Such usage currently requires loading SUN provider which currently always allows to calculate MD5 digest.
+
+Chainguard is working on enabling to jlink BouncyCastle FIPS into the runtime image, to eliminate SUN dependency, then it would be possible to control MD5 access with bouncycastle specific security properties. This guidance will be updated when the situation changes.
+
+If blocking behaviour is required, ensure to request implementation from BCFIPS provider, or check for BCFIPS provider, as it has higest priority in the bcfips hardened configuration.
```

Comment:
`s/behaviour/behavior/` - use English (Simplified) spelling
`s/BCFIPS/the BouncyCastle FIPS/g`

Replace "bcfips hardened" with whatever the BouncyCastle FIPS hardened configuration is named (presumably like `openssl-confg-fipshardened`?) so they know what to use.